packages/mongodb-memory-server-core/src/util/MongoBinary.ts

```diff
@@ -147,30 +164,40 @@ export default class MongoBinary {
           .split('\n')[0]
           .split(' ')[2];
 
-        if (version !== LATEST_VERSION && version !== binaryVersion) {
+        if (options.version !== LATEST_VERSION && options.version !== binaryVersion) {
           // we will log the version number of the system binary and the version requested so the user can see the difference
           log(
             'MongoMemoryServer: Possible version conflict\n' +
               `  SystemBinary version: ${binaryVersion}\n` +
-              `  Requested version:    ${version}\n\n` +
+              `  Requested version:    ${options.version}\n\n` +
               '  Using SystemBinary!'
           );
         }
       }
     }
 
     if (!binaryPath) {
-      binaryPath = await this.getCachePath(version);
+      binaryPath = this.getCachePath(options.version);
     }
 
     if (!binaryPath) {
       binaryPath = await this.getDownloadPath(options);
     }
 
-    log(`MongoBinary: Mongod binary path: ${binaryPath}`);
+    if (!binaryPath) {
+      throw new Error(
+        `MongoBinary.getPath: could not find an valid binary path! (Got: "${binaryPath}")`
+      );
+    }
+
+    log(`MongoBinary: Mongod binary path: "${binaryPath}"`);
     return binaryPath;
   }
 
+  /**
+   * Purpose unkown, not used anywhere
+   * @param files Array of Strings
+   */
   static hasValidBinPath(files: string[]): boolean {
```

Comment:
still dont know what this function is for (#373)

Reply:
I removing this ;)
Some old code 🤗